### PR TITLE
Find ROCm device library in conda layout

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/patches/0009-Find-ROCm-device-library-in-conda-layout.patch [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
++  llvm::sys::fs::real_path(D.getClangProgramPath(), RealClangPath);
++  LibDevicePath = llvm::sys::path::parent_path(RealClangPath);
++  llvm::sys::path::append(LibDevicePath, CLANG_INSTALL_LIBDIR_BASENAME,
++                          "amdgcn", "bitcode");
++  HasDeviceLibrary = CheckDeviceLib(LibDevicePath, true);
++  if (HasDeviceLibrary)
++    return;
++
+   // Find device libraries in a legacy ROCm directory structure
+   // ${ROCM_ROOT}/amdgcn/bitcode/*
```
Comment (h-vetinari) [inline recipe/patches/0009-Find-ROCm-device-library-in-conda-layout.patch:38]: Is there a reason why we couldn't/shouldn't just set `ROCM_ROOT=$BUILD_PREFIX/lib` (or wherever relative to where our clang sits)?

Comment (isuruf) [reply recipe/patches/0009-Find-ROCm-device-library-in-conda-layout.patch:38]: Yes, because then `ROCM_ROOT/bin` would be `$BUILD_PREFIX/lib/bin`

Comment (h-vetinari) [reply recipe/patches/0009-Find-ROCm-device-library-in-conda-layout.patch:38]: OK, looking at the surrounding [context](https://github.com/llvm/llvm-project/blob/release/17.x/clang/lib/Driver/ToolChains/AMDGPU.cpp#L413-L429), this makes sense; we're not gonna have the rocm-bits in a clang specific path.

**Review by h-vetinari — APPROVED** [review]

**Comment by isuruf** [comment]
Thanks for the review.

**Pull request merged**